### DISCUSSION TITLE
Lab6 韌體與hex問題

### DISCUSSION THREAD

**Discussion by BarryJu1999** [bol-edu/HLS-SOC-Discussions#168] [Q&A]
各位好，目前lab6卡在無法跑完韌體，
目前嘗試了幾種辦法，

1.跑除了fir的其他三個功能：                          成功

2.跑fir+uart：                                                  失敗，卡在ab40(flag start)

3.跑四個功能但fir不做運算直接輸出input：   成功

4.同上但改fir.c中outputsample = input*tap：失敗，卡在ab40(flag start)

5.同上但改fir.c中outputsample = tap：          成功

以上看起來問題應該不是硬體，
而是在於fir不能做運算或是運算過程轉hex會有問題，想請問如何解決?

**Comment by LingChiYang** [comment]
同學好

是用同學自己的FIR嗎?

**Reply by BarryJu1999** [reply]
並不是
我是使用這次lab中的fir.c做使用(下圖)
![螢幕擷取畫面 2023-12-06 180809](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144991650/4cc89178-c929-4416-b89f-a00cc50daf6d)

並且在測試中我們發現只用加法實作是可以繞過這個問題，
但當我們要整合其他功能時又會遇到相同狀況卡在ab40，
另外當我們的pointer都使用同一個時(不宣告三個pointer給三個功能)，
韌體可以在往下執行一點點(不是flag start)，但仍舊是卡在中間
因此我們目前認為是記憶體問題但又無從下手。

**Reply by LingChiYang** [reply]
模擬有跑完嗎? 
從這張圖我無法判斷是什麼問題，因為矩陣乘法也有用乘法，我們測試記憶體也完全夠，方便給我你們的repo嗎?

**Reply by BarryJu1999** [reply]
https://github.com/BarryJu1999/SOC_LAB6

資料夾為testbench的內容，波型檔好像太大放不上來，
裡面有fir_nomul即我們修改後無乘法的版本，
另外wrapper_all.v即為本次的硬體設計，
模擬是有跑完的。
再麻煩助教了。

**Reply by LingChiYang** [reply]
同學好

複製你上述說的main是有跑到AB54的，你可能要提供user_project其他檔案以及其他有修改firmware的細節
